lib/recognize-stream.ts

```diff
@@ -398,6 +402,12 @@ class RecognizeStream extends Duplex {
       // can't send any more data after the stop message (although this shouldn't happen normally...)
       return;
     }
+
+    // if using iam, we need to authenticate the first time `_write` is called
+    if (!this.authenticated) {
+      return this.setAuthorizationHeaderToken(callback);
```

Comment:
If you call `_write()` and you don't write it to the stream you may lose your first bytes... I think you need to have a callback to check if there is a token and then operates like it used to do.

You can't call the same callback more than once and/or ignore the first bytes.